diesel_cli/src/cli.rs

```diff
@@ -90,8 +90,18 @@ pub fn build_cli() -> App<'static, 'static> {
         )
         .setting(AppSettings::SubcommandRequiredElseHelp);
 
-    let generate_bash_completion_subcommand = SubCommand::with_name("bash-completion")
-        .about("Generate bash completion script for the diesel command.");
+    let generate_completions_subcommand = SubCommand::with_name("completions")
```

Comment:
What about making the shell variants just arguments?
Something like this:

```rust
SubCommand::with_name("completions")
    .about("Generate shell completion scripts for the diesel command.")
    .arg(Arg::with_name("SHELL")
        .index(1)
        .required(true)
        .possible_values(&Shell::variants())
    .setting(AppSettings::SubcommandRequiredElseHelp);

```

This would simplify the implementation and automatically support all shells that are added in a future version of clap.

Reply:
I somehow missed that function, I'm now using your snippet and also found the `value_t!` macro, this removes every reference to bash or zsh and generates completions for everything supported in clap.

Reply:
Removing the `bash-completion` command is a breaking change, and shouldn't be done in a 1.x version. We need to keep that command and print a deprecation warning.

Reply:
@sgrif I've re-introduced the command and also added a deprecation warning to stderr.